### ISSUE TITLE
Add defer to gopher Lua

### ISSUE THREAD

**Issue by dukv2001** [yuin/gopher-lua#299]
- [ yes] GopherLua is a Lua5.1 implementation. You should be familiar with Lua programming language. Have you read [Lua 5.1 reference manual](http://www.lua.org/manual/5.1/) carefully? 
- [ yes] GopherLua is a Lua5.1 implementation.  In Lua, to keep it simple, **it is more important to remove functionalities rather than to add functionalities unlike other languages** . If you are going to introduce some new cool functionalities into the GopherLua code base and the functionalities can be implemented by existing APIs, It should be implemented as a library. 

Please answer the following before submitting your issue:

1. What version of GopherLua are you using? : 5.1
2. What version of Go are you using? : 1.15
3. What operating system and processor architecture are you using? :windows/x64
4. What did you do? :add lua defer
5. What did you expect to see? : defer in lua
6. What did you see instead? : no

function foo()
    defer
        print 'defer 1'
    end

    do
        defer print 'defer 2' end
    end

    defer print 'defer 3' end

    print 'end'
end

https://github.com/peete-q/lua-defer

**Comment by rhettli** [comment]
yeah,i like this function,it is so cool.

**Comment by dukv2001** [comment]
In order to adapt to intelligent grammar tips, I think we should use the following grammar

defer(function()
  print("call")
end)

Lua needs him so much

**Comment by rhettli** [comment]
> In order to adapt to intelligent grammar tips, I think we should use the following grammar
> 
> defer(function()
> print("call")
> end)
> 
> Lua needs him so much



> In order to adapt to intelligent grammar tips, I think we should use the following grammar
> 
> defer(function()
> print("call")
> end)
> 
> Lua needs him so much


yeah ,without defer , program may not close mysql handler when got an error:

**Comment by rhettli** [comment]
@dukv2001 but as you know , gopher-lua is build with go fully, without any other language code, maybe it is  diffcat~  to put this function to it

**Comment by dukv2001** [comment]
No, it needs to be supported in the language. External functions can't

**Comment by a97077088** [comment]
已经实现了这个功能，并且带智能提示
